internal/controllers/topology/cluster/desired_state.go

```diff
@@ -897,19 +908,29 @@ func templateToTemplate(in templateToInput) *unstructured.Unstructured {
 	// Enforce the topology labels into the provided label set.
 	// NOTE: The cluster label is added at creation time so this object could be read by the ClusterTopology
 	// controller immediately after creation, even before other controllers are going to add the label (if missing).
-	labels := template.GetLabels()
-	if labels == nil {
-		labels = map[string]string{}
+	labels := map[string]string{}
+	for k, v := range in.labels {
+		labels[k] = v
+	}
+	if template.GetLabels() != nil {
+		for k, v := range template.GetLabels() {
+			labels[k] = v
+		}
 	}
```

Comment:
```suggestion
	labels := template.GetLabels()
	if labels == nil {
		labels = map[string]string{}
	}
	for k, v := range in.labels {
		labels[k] = v
	}
```

What about this way? I think the labels from in.labels should have higher priority (just like e.g. in GenerateTemplate

Reply:
(same for annotations)